import "base.proto";

package dist_clang.proto;

message Host {
  required string host          = 1;
  optional uint32 port          = 2 [default=6000];
  optional uint32 threads       = 3 [default=2];
  optional bool disabled        = 4 [default=false];
}

message Verbosity {
  message Range {
    required uint32 right = 1;
    optional uint32 left  = 2;  // if omitted, then range contains one number.
  }

  // less'o'equal to |error_mark| levels go to stderr.
  optional uint32 error_mark    = 1 [default=20];

  // all ranges are inclusive: [left, right].
  repeated Range levels    = 2;
}

message Configuration {
  optional string socket_path   = 1;
  optional uint32 pool_capacity = 2 [default=16];  // 0 - is unlimited.
  optional string cache_path    = 3;
  repeated Host remotes         = 4;
  optional Host local           = 5;
  repeated Compiler versions    = 6;
  optional Host statistic       = 7;
  optional Verbosity verbosity  = 8;
  optional uint64 cache_size    = 9 [default=0];  // 0 - is unlimited.
  optional uint32 user_id       = 10;
  optional bool remote_cache    = 11 [default=true];
  optional bool sync_cache      = 12 [default=false];  // for testing.
}
